src/main/java/seedu/address/logic/parser/BorrowCommandParser.java

```diff
@@ -30,11 +30,8 @@ public BorrowCommand parse(String args) throws ParseException {
         // Ensures date can be parsed because regex does not check for all invalid cases.
         // Check is done here instead of BookStatus class so that invalid date error is not
         // thrown from BookStatus, therefore hiding BookStatus from end user.
-        try {
-            STATUS_DATE_FORMAT.parse(returnDate);
-        } catch (java.text.ParseException e) {
-            throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, BorrowCommand.MESSAGE_USAGE));
+        if (!isValidDateString(returnDate)) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, BorrowCommand.MESSAGE_USAGE));
```

Comment:
Good job with outputting the relevant error message when an invalid date is passed 👍